### PR TITLE
[SDPA-4225] updates tide_site to be compatible with Drupal 8.8.5

### PR DESCRIPTION
**Pull request by vincent-gao**
### Jira
https://digital-engagement.atlassian.net/browse/SDPA-4225
### Changes
1. [The Path Alias core subsystem has been moved to the "path_alias" module](https://www.drupal.org/node/3092086)
   - refactor path alias 
2. [Resource (entity) normalization should use partial caching](https://www.drupal.org/project/drupal/issues/2819335)
   - override jsonapi `ResourceObjectNormalizationCacher` to include `url.query_args:site` cache context.
3. https://github.com/dpc-sdp/tide_core/pull/139
4. https://github.com/dpc-sdp/dev-tools/pull/36
5. https://github.com/dpc-sdp/tide_test/releases/tag/1.2.0
6. https://github.com/dpc-sdp/content-vic-gov-au/pull/867

### CODE REVIEW

**Comment by vincent-gao** [comment]
The changes has been passed CI, 
more details see 
https://circleci.com/gh/dpc-sdp/tide_site/958
https://circleci.com/gh/dpc-sdp/tide_site/959

File: src/AliasStorageHelper.php [hunk 1]
```diff
@@ -359,9 +340,35 @@ public function uniquify(&$alias, $langcode) {
     do {
       // Append an incrementing numeric suffix until we find a unique alias.
       $unique_suffix = $separator . $i;
-      $alias = Unicode::truncate($original_alias, $maxlength - Unicode::strlen($unique_suffix), TRUE) . $unique_suffix;
+      $alias = Unicode::truncate($original_alias, $maxlength - mb_strlen($unique_suffix), TRUE) . $unique_suffix;
```
[the remark this answers is not in the document]
Comment (vincent-gao) [reply src/AliasStorageHelper.php:343]: yes, no problem. @anthony-malkoun

File: src/AliasStorageHelper.php [hunk 2]
```diff
@@ -269,30 +274,6 @@ public function updateSiteAliases($path, $original_path) {
     }
   }
 
-  /**
-   * Delete all site copies of a path alias.
-   *
-   * @param array|bool $path
-   *   The Path array.
-   */
-  public function deleteSiteAliases($path) {
```
[the remark this answers is not in the document]
Comment (vincent-gao) [reply src/AliasStorageHelper.php:278]: @anthony-malkoun Yes. I think so.

File: src/LinkitResultManager.php [hunk 3]
```diff
@@ -30,27 +22,29 @@ class LinkitResultManager extends ResultManager {
...
    */
-  public function getResults(ProfileInterface $linkitProfile, $search_string) {
-    $matches = parent::getResults($linkitProfile, $search_string);
-    foreach ($matches as &$match) {
-      $path = $this->aliasStorage->load(['alias' => $match['path']]);
-      if ($path) {
-        $node = $this->aliasHelper->getNodeFromPath($path);
-        if ($node) {
-          $match['path'] = '/node/' . $node->id();
+  public function getSuggestions(ProfileInterface $linkitProfile, $search_string) {
```
[the remark this answers is not in the document]
Comment (vincent-gao) [reply src/LinkitResultManager.php:32]: @anthony-malkoun Thanks

File: tide_site.module [hunk 4]
```diff
@@ -336,17 +338,16 @@ function tide_site_tide_path_enhancer_undo_transform_alter(&$data, &$context) {
...
   $data['origin_alias'] = $data['alias'];
   $data['alias'] = $alias_helper->getPathAliasWithoutSitePrefix($data);
 
-  $path = $alias_storage->load(['pid' => $data['pid']]);
-  if ($path) {
-    $node = $alias_helper->getNodeFromPath($path);
+  /** @var \Drupal\path_alias\Entity\PathAlias $path_entity */
+  $path_entity = PathAlias::load($data['pid']);
+  if ($path_entity) {
+    $node = $alias_helper->getNodeFromPath($path_entity);
```
[the remark this answers is not in the document]
Comment (vincent-gao) [reply tide_site.module:350]: @anthony-malkoun Yes, we should. I will do some changes.

File: tide_site.services.yml [hunk 5]
```diff
@@ -1,10 +1,7 @@
 services:
-  tide_site.alias_storage:
-    class: Drupal\tide_site\AliasStorage
-    arguments: ['@database', '@module_handler']
   tide_site.alias_storage_helper:
     class: Drupal\tide_site\AliasStorageHelper
-    arguments: ['@tide_site.helper', '@tide_site.alias_storage', '@entity_type.manager']
```
[the remark this answers is not in the document]
Comment (vincent-gao) [reply tide_site.services.yml:7]: Yes. @anthony-malkoun . we dont need `tide_site.alias_storage` as path aliases have been converted to revisionable entities. `tide_site.alias_storage` or its parent were provide `CRUD` functions for path alias table.
now, however we are using `\Drupal::entityTypeManager()->getStorage('path_alias')->load($id)` to get the path entity.

File: src/AliasStorageHelper.php [hunk 6]
```diff
@@ -359,9 +340,35 @@ public function uniquify(&$alias, $langcode) {
     do {
       // Append an incrementing numeric suffix until we find a unique alias.
       $unique_suffix = $separator . $i;
-      $alias = Unicode::truncate($original_alias, $maxlength - Unicode::strlen($unique_suffix), TRUE) . $unique_suffix;
+      $alias = Unicode::truncate($original_alias, $maxlength - mb_strlen($unique_suffix), TRUE) . $unique_suffix;
       $i++;
     } while ($this->isAliasExists($alias, $langcode));
   }
 
+  /**
```
[the remark this answers is not in the document]
Comment (vincent-gao) [reply src/AliasStorageHelper.php:348]: @anthony-malkoun yeah. i will try to delete it. Thanks @anthony-malkoun

**Comment by vincent-gao** [comment]
> This looks mostly good @vincent-gao. Well done on working out the changes to using path entities. I have a few questions and wondering why CI isn't passing. I tried on my local and got the same error.

1. @anthony-malkoun it passes the CI building and test. https://app.circleci.com/pipelines/github/dpc-sdp/tide_site/46/workflows/eb91c6da-5d3b-4794-b2d5-cc4b8691585c
I dont know why https://github.com/dpc-sdp/tide_site/pull/55 UI shows both of them get failed. 
2. I will try do some changes based on your comments.

File: tide_site.module [hunk 7]
```diff
@@ -500,77 +499,60 @@ function tide_site_form_alter(array &$form, FormStateInterface $form_state, $for
...
-        /** @var \Drupal\tide_site\AliasStorage $alias_storage */
-        $alias_storage = \Drupal::service('tide_site.alias_storage');
-        $alias_storage->delete(['pid' => $path['pid']], FALSE);
+        $path->delete();
       }
     }
   }
 }
 
-/**
```
[the remark this answers is not in the document]
Comment (vincent-gao) [reply tide_site.module:550]: @anthony-malkoun yes. IMO we dont need it anymore.
first of all, I cannot convince myself to implement this hook here. 
I think the changing of alias is based on the changing of nodes, and the CI tests and regression tests show that no page errors, logic errors  and no alias errors. so we don't need the hook_path_alias_delete().
Thoughts?

[the remark this answers is not in the document]
Comment (anthony-malkoun) [reply tide_site.module:550]: But I think the point is if you've created a node for multiple sites and you delete an alias, do you need to delete all the alias for the other sites? e.g. if you have `/site-4/demo-content` do you want to also delete `/site-223/demo-content` automatically?

**Comment by vincent-gao** [comment]
> Thanks for working through this @vincent-gao. I think we'll need to be aware of the edge cases of updating the existing aliases etc. Can you add this to the acceptance criteria in the ticket please so we can make sure it gets tested. Approved.

yes. sure! definitely. will do @anthony-malkoun

**Pull request merged**